src/uu/sort/src/sort.rs

```diff
@@ -322,25 +323,52 @@ struct Precomputed {
 
 impl GlobalSettings {
     /// Parse a SIZE string into a number of bytes.
-    /// A size string comprises an integer and an optional unit.
+    /// A size string comprises an integer and an optional unit or percentage of total memory.
     /// The unit may be k, K, m, M, g, G, t, T, P, E, Z, Y (powers of 1024), or b which is 1.
     /// Default is K.
     fn parse_byte_count(input: &str) -> Result<usize, ParseSizeError> {
         // GNU sort (8.32)   valid: 1b,        k, K, m, M, g, G, t, T, P, E, Z, Y
         // GNU sort (8.32) invalid:  b, B, 1B,                         p, e, z, y
+        let input = input.trim();
+        if let Some(number) = input.strip_suffix('%') {
+            let percent = Self::parse_percentage(number)?;
+            return Self::get_memory_size(percent);
+        }
         let size = Parser::default()
             .with_allow_list(&[
                 "b", "k", "K", "m", "M", "g", "G", "t", "T", "P", "E", "Z", "Y",
             ])
             .with_default_unit("K")
             .with_b_byte_count(true)
-            .parse(input.trim())?;
+            .parse(input)?;
 
         usize::try_from(size).map_err(|_| {
             ParseSizeError::SizeTooBig(format!("Buffer size {size} does not fit in address space"))
         })
     }
 
+    /// Parses percentage input from input string.
+    fn parse_percentage(number_str: &str) -> Result<usize, ParseSizeError> {
+        let Ok(percent) = number_str.parse::<u64>() else {
+            return Err(ParseSizeError::ParseFailure(format!(
+                "Invalid Buffer Percentage: {number_str}"
+            )));
+        };
+        match percent {
+            0..=100 => Ok(percent as usize),
+            _ => Err(ParseSizeError::ParseFailure(format!(
+                "Invalid Buffer Percentage: {number_str}"
+            ))),
+        }
+    }
+    /// get memory size from percent
+    fn get_memory_size(percent: usize) -> Result<usize, ParseSizeError> {
```

Comment:
```suggestion
    fn get_memory_size(percent: usize) -> usize {
```
As far as I can tell there's no way this can fail right?

Reply:
@zeroishero ping?

Reply:
Forgot about this. I will try to complete this soon